vaadin-button-flow-parent/vaadin-button-flow/src/main/resources/META-INF/resources/frontend/buttonFunctions.js

```diff
@@ -0,0 +1,12 @@
+window.Vaadin.Flow.button = {
+  initDisableOnClick: (button) => {
+    if (!button.__hasDisableOnClickListener) {
+      button.addEventListener('click', disableOnClickListener(button));
```

Comment:
Shouldn't run the function here but pass the function as the click listener

```suggestion
      button.addEventListener('click', disableOnClickListener);
```